src/test/multisig_tests.cpp

```diff
@@ -203,7 +203,10 @@ BOOST_AUTO_TEST_CASE(multisig_IsStandard)
     malformed[5] << OP_1 << ToByteVector(key[0].GetPubKey()) << ToByteVector(key[1].GetPubKey());
 
     for (int i = 0; i < 6; i++)
-        BOOST_CHECK(!::IsStandard(malformed[i], whichType));
+    {
+        BOOST_CHECK(::IsStandard(malformed[i], whichType));
+        BOOST_CHECK(whichType == TX_CUSTOM);
+    }
```

Comment:
Please not this case - malformed multisig scripts are also accepted in TX_CUSTOM category. Should we check this more strictly?

Reply:
Improved this in the new commits - Now I try to identify the type of script (p2pkh, p2sh and multis) from the script op code and flag it as nonstandard before checking disallowed opcodes to tag the script as custom.

Reply:
malformed multisig scripts should be accepted. we do not check it.

Reply:
Then I'll revert back the newer commits. it will restore the expected behaviour.